extensions/interactions/TextInput/TextInputPredictionServiceSpec.js

```diff
@@ -0,0 +1,55 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests text input prediction service.
+ */
+
+describe('Text Input Prediction Service', function() {
+  var $rootScope = null;
+  var $scope = null;
+
+  beforeEach(module('oppia'));
+
+  describe('Test text prediction service', function() {
+    var predictionService;
+    beforeEach(inject(function($injector) {
+      predictionService = $injector.get('TextInputPredictionService');
+      $rootScope = $injector.get('$rootScope');
+      $scope = $rootScope.$new();
+    }));
+
+    it('should predict correct answer group for the answers', function() {
+      // These answers are taken from the text_input_training_data.json
+      // in Oppia-ml. Never test classifiers using training data unless it
+      // is only the functionality that you want to test (like in this case).
+
+      jasmine.getJSONFixtures().fixturesPath = 
```

Comment:
FYI, "/anmol" probably shouldn't appear in this PR :)

I remember doing stuff with fixtures before. Take a look at extensions/interactions/rulesSpec.js (and extensions/interactions/rule_templates.json in karma.conf.js).

Reply:
Haha yep, this one is strictly WIP. There are so many hacks that you might get a heart-attack if you review it now :stuck_out_tongue: 

Thanks, the path should actually be relative to the `base` dir. I'll change it (and fix everything else) once we have addressed the more important issue here - fixing the prediction service.